### PR TITLE
feat: clean up pipeline archives

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
clean up pipeline archives that have exceeded default retention time

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=248988&iterationID=-1&pId=290591&type=TASK)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Support clean up pipeline archives           |
| 🇨🇳 中文    |  实现了pipeline归档表的自动清理            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .erda/migrations/pipeline/20220322-pipeline-archive-add-index.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `pipeline_archives` ADD INDEX `idx_status_timecreated` (`status`, `time_created`);
```
Comment (sfwn) [inline .erda/migrations/pipeline/20220322-pipeline-archive-add-index.sql:1]: time_created or time_end
you can add index for both fields.

File: modules/pipeline/providers/dbgc/db/pipeline_archives.go [hunk 2]
```diff
@@ -16,14 +16,22 @@ package db
...
 
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 
 	"github.com/erda-project/erda-infra/base/version"
 	"github.com/erda-project/erda/modules/pipeline/dbclient"
 	"github.com/erda-project/erda/modules/pipeline/spec"
 )
 
+type DeleteArchiveRequest struct {
```
Comment (sfwn) [inline modules/pipeline/providers/dbgc/db/pipeline_archives.go:29]: ArchiveDeleteRequest
like PipelineCreateRequest

File: modules/pipeline/providers/dbgc/provider.go [hunk 3]
```diff
@@ -28,7 +29,12 @@ import (
 	"github.com/erda-project/erda/pkg/jsonstore/etcd"
 )
 
-type config struct{}
+type config struct {
+	// 7 * 24 * 60 * 60
+	AnalyzedPipelineArchiveDefaultRetainSec time.Duration `file:"analyzed_pipeline_archive_default_retain_sec" default:"604800s"`
```
Comment (sfwn) [inline modules/pipeline/providers/dbgc/provider.go:34]: use provider.config

File: modules/pipeline/providers/dbgc/dbgc.go [hunk 4]
```diff
@@ -57,6 +60,9 @@ func (p *provider) PipelineDatabaseGC(ctx context.Context) {
 
 			p.doNotAnalyzedPipelineDatabaseGC(true)
 			p.doNotAnalyzedPipelineDatabaseGC(false)
+
+			p.doAnalyzedPipelineArchiveGC()
```
Comment (sfwn) [inline modules/pipeline/providers/dbgc/dbgc.go:64]: ugly.
combine with L50.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/dbgc/provider.go (hunk 3 above)
Comment (chengjoey) [reply modules/pipeline/providers/dbgc/provider.go:34]: image

File: .erda/migrations/pipeline/20220322-pipeline-archive-add-index.sql (hunk 1 above)
Comment (chengjoey) [reply .erda/migrations/pipeline/20220322-pipeline-archive-add-index.sql]: added

File: modules/pipeline/providers/dbgc/db/pipeline_archives.go (hunk 2 above)
Comment (chengjoey) [reply modules/pipeline/providers/dbgc/db/pipeline_archives.go]: adjusted

File: modules/pipeline/providers/dbgc/dbgc.go (hunk 4 above)
Comment (chengjoey) [reply modules/pipeline/providers/dbgc/dbgc.go]: combine `doAnalyzedPipelineArchiveGC` and `doNotAnalyzedPipelineArchiveGC`

File: modules/pipeline/providers/dbgc/provider.go [hunk 5]
```diff
@@ -28,7 +29,12 @@ import (
...
+	// 30 * 24 * 60 * 60
+	FinishedPipelineArchiveDefaultRetainSec time.Duration `file:"finished_pipeline_archive_default_retain_sec" default:"2592000s"`
```
Comment (sfwn) [inline modules/pipeline/providers/dbgc/provider.go:36]: use `720h` is better.

File: modules/pipeline/providers/dbgc/provider.go (hunk 3 above)
Comment (sfwn) [inline modules/pipeline/providers/dbgc/provider.go:34]: use hour is better.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/dbgc/provider.go (hunk 5 above)
Comment (chengjoey) [reply modules/pipeline/providers/dbgc/provider.go]: adjusted

File: modules/pipeline/providers/dbgc/provider.go (hunk 3 above)
Comment (chengjoey) [reply modules/pipeline/providers/dbgc/provider.go]: adjusted

**Review by sfwn — APPROVED** [review]

**Pull request merged**